parser/rootfs_parser.go

```diff
@@ -0,0 +1,346 @@
+// Copyright 2016 Mender Software AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package parser
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"crypto/sha256"
+	"encoding/hex"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/mendersoftware/artifacts/archiver"
+	"github.com/mendersoftware/artifacts/metadata"
+	"github.com/mendersoftware/log"
+	"github.com/pkg/errors"
+)
+
+type rootfsFile struct {
+	name      string
+	path      string
+	size      int64
+	date      time.Time
+	checksum  []byte
+	signature []byte
+}
+
+type RootfsParser struct {
+	files    metadata.Files
+	tInfo    metadata.TypeInfo
+	metadata metadata.Metadata
+	updates  map[string]rootfsFile
+
+	sStore string
+	dStore io.Writer
+}
+
+func NewRootfsParser(sStoreDir string, w io.Writer) RootfsParser {
+	return RootfsParser{
+		sStore:  sStoreDir,
+		dStore:  w,
+		updates: map[string]rootfsFile{}}
+}
+
+func (rp RootfsParser) NeedsDataFile() bool {
+	return true
+}
+
+func (rp *RootfsParser) ArchiveData(tw *tar.Writer, srcDir, dst string) error {
+	f, err := os.Create("/tmp/my_data.tar.gz")
```

Comment:
Why writing to tmpfs ? isn't this equivalent as keeping this in memory ?
What with about concurrent access if we have 2 instances of RootfsParser working on something?

For intermediate files this is useful: https://golang.org/pkg/io/ioutil/#TempFile